# Selection of CVD-associated SNPs
```{r}
gwas <- read_tsv("gwas_catalog_v1.0-associations_e109_r2023-05-20.tsv")
``` 
# Cardiac filter
```{r}
cardiac_gwas <- gwas %>%
  select('DISEASE/TRAIT', REGION, CHR_ID, CHR_POS, MAPPED_GENE, UPSTREAM_GENE_ID, DOWNSTREAM_GENE_ID, SNP_GENE_IDS, SNPS, CONTEXT, STUDY) %>%
  filter(grepl('heart|cardiac|aortic|atrial|ventric|cardio|vascular|artery|coronary|myocardial|valve|cardio|cardium|stroke', `DISEASE/TRAIT`))
```

# CVD-SNPs per Chromosome
```{r}
library(ggplot2)
```

# Calculate the count of SNPs in each chromosome
```{r}
SNP_counts <- table(CVD_SNP_count$chromosome)
```

# Create a data frame with the count information
```{r}
count_data <- data.frame(chromosome = as.numeric(names(SNP_counts)), count = as.numeric(SNP_counts))
```

# Define the desired color palette
```{r}
color_palette <- c("#4197d8", "#f8c120", "#413496", "#495226",
                   "#d60b6f", "#e66519", "#d581b7", "#83d3ad",
                   "#7c162c", "#26755d", rep(color_palette_all[1:10], 2))
```

# Create the bar plot
```{r}
ggplot(count_data, aes(x = chromosome, y = count, fill = as.factor(chromosome))) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = color_palette) +
  labs(x = "Chromosome", y = "SNP Count") +
  ggtitle("SNP Count by Chromosome") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "none",
        panel.background = element_rect(fill = "white"),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(color = "black")) +
  scale_x_continuous(breaks = count_data$chromosome)
```

# Cardiac eQTL
```{r}
CVD_genes <- read_tsv("CVD_genes.txt")
```

# Convert chromosome column to numeric
```{r}
CVD_genes$CHR <- as.numeric(gsub("chr", "", CVD_genes$CHR))
```

# Sort the dataframe by chromosome and position
```{r}
CVD_genes <- CVD_genes[order(CVD_genes$CHR, CVD_genes$BP), ]
```

```{r}
library(qqman)
```

# Create the Manhattan plot
```{r}
manhattan_plot <- manhattan(CVD_genes, col = c("#4197d8", "#f8c120", "#413496", "#495226",
                                               "#d60b6f", "#e66519", "#d581b7", "#83d3ad",
                                               "#7c162c", "#26755d"), ylim = c(0, max(-log10(CVD_genes$P))*1.1))
```

# SNP Frequency per Chromosome Mb
## SNP Density CMplot

```{r}
library(CMplot)
SNP_Density<-read_tsv("Manhattan plot.txt")

CMplot(SNP_Density, col=c("#4197d8", "#f8c120", "#413496", "#495226",
    "#d60b6f", "#e66519", "#d581b7", "#83d3ad", "#7c162c", "#26755d"),
    bin.size=1e6, bin.range=NULL, bin.legend.num=10, pch=19, type="p",
    band=1, H=1.5, ylim=NULL, cex.axis=1, lwd.axis=1.5, cex.lab=1.5,
    plot.type="b", multracks=FALSE, points.alpha=100L, cex=c(0.5,1,1),
    r=0.3, outward=FALSE, ylab=expression(-log[10](italic(p))),
    ylab.pos=3, xticks.pos=1, mar = c(3,6,3,3), threshold = NULL,
    threshold.col="red", threshold.lwd=1, threshold.lty=2,
    amplify= TRUE, signal.cex = 1.5, signal.pch = 19,
    signal.col=NULL, signal.line=2, highlight=NULL, highlight.cex=1,
    highlight.pch=19, highlight.type="p", highlight.col="red",
    highlight.text=NULL, highlight.text.col="black", highlight.text.cex=1,
    highlight.text.xadj=NULL, highlight.text.yadj=NULL,
    highlight.text.font=3, chr.labels=NULL, chr.border=FALSE,
    chr.labels.angle=0, chr.den.col="black", chr.pos.max=FALSE, cir.band=1,
    cir.chr=TRUE, cir.chr.h=1.5, cir.legend=TRUE, cir.legend.cex=0.6,
    cir.legend.col="black", LOG10=TRUE, box=FALSE, conf.int=TRUE,
    conf.int.col=NULL, file.output=TRUE, file.name="", file=c("jpg","pdf"
    ,"tiff"), dpi=300, height=NULL, width=NULL, main="", main.cex=1.5,
    main.font=2, trait.legend.ncol=NULL, trait.legend.cex=NULL,
    trait.legend.pos=c("left","middle","right"),verbose=TRUE)
```